horovod/spark/data_loaders/pytorch_data_loaders.py

```diff
@@ -95,14 +96,60 @@ def _iterate(self):
                 break
             num_steps += 1
 
-            yield next(self.iterater)
+            yield next(self.iterator)
 
 
 class PytorchInfiniteAsyncDataLoader(AsyncDataLoaderMixin, PytorchInfiniteDataLoader):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
 
+class PytorchInmemDataLoader(BaseDataLoader):
+    def __init__(self, reader, batch_size, num_epochs, name="",
+                 shuffle=False, limit_step_per_epoch=-1, verbose=False):
+        self.batch_size = batch_size
+        self.limit_step_per_epoch = limit_step_per_epoch
+        self.name = name
+        self.verbose = verbose
+
+        if limit_step_per_epoch == -1:
+            raise ValueError('limit_step_per_epoch cannot be -1 for inmem dataloader')
+
+        print(f"[{self.name}]: Initializing petastorm inmem_dataloader with batch_size={batch_size}"
+              f"num_epochs={num_epochs}, "
+              f"shuffle={shuffle}"
+              f"limit_step_per_epoch={limit_step_per_epoch}")
+
+        self.dataloader = InMemBatchedDataLoader(reader, batch_size=batch_size, num_epochs=num_epochs,
+                                                 rows_capacity=batch_size*limit_step_per_epoch, shuffle=shuffle)
+        self.iterator = iter(self.dataloader)
+
+    def __len__(self):
+        # We cannot infer length from reader.
+        return self.limit_step_per_epoch if self.limit_step_per_epoch != -1 else 0
```

Comment:
Don't we raise an exception if `self.limit_step_per_epoch == -1`? Can we then just have this always return `self.limit_step_per_epoch`?